configure.ac

```diff
@@ -2968,36 +2968,11 @@ AS_IF(test "x$enable_zph_qos" = "xyes"],[
 ])
 
 AC_CHECK_LIB(regex, regexec, [REGEXLIB="-lregex"],[REGEXLIB=''])
-AC_ARG_ENABLE(gnuregex,
-  AS_HELP_STRING([--enable-gnuregex],
-                 [Compile GNUregex.  Unless you have reason to use
-                 this option, you should not enable it.
-                 This library file is usually only required on Windows and
-                 very old Unix boxes which do not have their own regex
-                 library built in.]), [
-  SQUID_YESNO([$enableval],[--enable-gnuregex])
-])
-# force-enable on old solaris
-AS_IF([test "x${enable_gnuregex:=auto}" = "xauto"],[
-  AS_CASE(["$host"],
-    [*-sun-solaris2.[[0-4]]],[enable_gnuregex="yes"]
-  )
-])
-
-# try detecting if it is needed
-AS_IF([test "x$enable_gnuregex" = "xauto"],[
-  SQUID_CHECK_REGEX_WORKS
-  AS_IF([test "x$squid_cv_regex_works" = "xyes"],[enable_gnuregex=no],[enable_gnuregex=yes])
-])
-AC_MSG_CHECKING(if GNUregex needs to be compiled)
-AC_MSG_RESULT($enable_gnuregex)
-AS_IF([test "x$enable_gnuregex" = "xyes"],[
-  # for some reason (force-enable, test..) gnuregex was found as needed. Override any system lib
-  REGEXLIB=""
+SQUID_CHECK_REGEX_WORKS
+AS_IF([test "x$squid_cv_regex_works" = "xno"],[
+  AC_MSG_ERROR([Could not detect a functioning regex library.
+                If your OS does not provide one, please install gnulib and retry the build])
```

Comment:
Please drop this and SQUID_CHECK_REGEX_WORKS itself. We do not need to (and, hence, should not half-)check, at ./configure time, that non-optional features compile (which is what  SQUID_CHECK_REGEX_WORKS does in this PR). We should rely on the compiler to detect and report any specific problems (when building Squid) instead of poorly duplicating that activity at ./configure time.

Reply:
Done. It would give us a chance to give a more meaningful error message, but in practice I don't think anyone would ever see it anyway

Reply:
> It would give us a chance to give a more meaningful error message

Actually, in many problematic cases, the error message would be more _meaningful_ when it happens during the primary build phase (or later), but this (debatable and varying) message quality is not the primary reason for removing this poor code duplication. Thank you for removing it!